File: src/StudentAttendanceRecordI.java
public class StudentAttendanceRecordI {
    public boolean checkRecord(String s) {
        int consecutiveLate = 0, absentees = 0;
        for (int index = 0 ; index < s.length() ; index++) {
            if (s.charAt(index) == 'L') {
                consecutiveLate++;
            } else {
                if (s.charAt(index) == 'A') absentees++;
                consecutiveLate = 0;
            }
            if (absentees >= 2 || consecutiveLate >= 3) {
                return false;
            }
        }
        return true;
    }
}

File: python/student_attendance_record_I.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        consecutiveLate, absent = 0, 0
        for character in s:
            if character == 'L':
                consecutiveLate += 1
            else:
                if character == 'A': absent += 1
                consecutiveLate = 0
            if consecutiveLate >= 3 or absent >= 2: return False
        return True
